Temporary files not cleared sometimes

This is observed in the flakiness of the integration test added in #252, e.g.:

```
FAILED tests/test_cvise.py::test_simple_reduction_no_interleaving_config - AssertionError: assert [PosixPath('/tmp/pytest-of-root/pytest-0/test_simple_reduction_no_inter0/tmpdir/ccT9pAPp.s'), PosixPath('/tmp/pytest-of-root/pytest-0/test_simple_reduction_no_inter0/tmpdir/ccSDGNXT.s')] == []
  
  Left contains 2 more items, first extra item: PosixPath('/tmp/pytest-of-root/pytest-0/test_simple_reduction_no_inter0/tmpdir/ccT9pAPp.s')
  
  Full diff:
  - []
  + [
  +     PosixPath('/tmp/pytest-of-root/pytest-0/test_simple_reduction_no_inter0/tmpdir/ccT9pAPp.s'),
  +     PosixPath('/tmp/pytest-of-root/pytest-0/test_simple_reduction_no_inter0/tmpdir/ccSDGNXT.s'),
  + ]
```

This might be similar to the problem previously reported in #144.

--- a/cvise/passes/abstract.py
+++ b/cvise/passes/abstract.py
@@ -122,7 +122,7 @@ class ProcessEventNotifier:
     def __init__(self, pid_queue):
         self.pid_queue = pid_queue
 
-    def run_process(self, cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=False):
+    def run_process(self, cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=False, env=None):
         if shell:
             assert isinstance(cmd, str)
         proc = subprocess.Popen(
@@ -132,6 +132,7 @@ class ProcessEventNotifier:
             universal_newlines=True,
             encoding='utf8',
             shell=shell,
+            env=env,
         )
         if self.pid_queue:
             self.pid_queue.put(ProcessEvent(proc.pid, ProcessEventType.STARTED))

--- a/cvise/utils/testing.py
+++ b/cvise/utils/testing.py
@@ -153,7 +153,7 @@ class TestEnvironment:
             # get cluttered with files it might leave undeleted (the process might do this because of an oversight in
             # the interestingness test, or because C-Vise abruptly kills our job without a chance for a proper cleanup).
             with tempfile.TemporaryDirectory(dir=self.folder, prefix='overridetmp') as tmp_override:
-                env = override_tmpdir_env(os.environ.copy(), tmp_override)
+                env = override_tmpdir_env(os.environ.copy(), Path(tmp_override))
                 stdout, stderr, returncode = ProcessEventNotifier(self.pid_queue).run_process(
                     str(self.test_script), shell=True, env=env
                 )

--- a/cvise/utils/testing.py
+++ b/cvise/utils/testing.py
@@ -16,7 +16,7 @@ import sys
 import tempfile
 import time
 import traceback
-from typing import Any, Callable, List, Union
+from typing import Any, Callable, List, Mapping, Union
 import concurrent.futures
 
 from cvise.cvise import CVise
@@ -149,9 +149,14 @@ class TestEnvironment:
     def run_test(self, verbose):
         try:
             os.chdir(self.folder)
-            stdout, stderr, returncode = ProcessEventNotifier(self.pid_queue).run_process(
-                str(self.test_script), shell=True
-            )
+            # Make the job use our custom temp dir instead of the standard one, so that the standard location doesn't
+            # get cluttered with files it might leave undeleted (the process might do this because of an oversight in
+            # the interestingness test, or because C-Vise abruptly kills our job without a chance for a proper cleanup).
+            with tempfile.TemporaryDirectory(dir=self.folder, prefix='overridetmp') as tmp_override:
+                env = override_tmpdir_env(os.environ.copy(), tmp_override)
+                stdout, stderr, returncode = ProcessEventNotifier(self.pid_queue).run_process(
+                    str(self.test_script), shell=True, env=env
+                )
             if verbose and returncode != 0:
                 logging.debug('stdout:\n' + stdout)
                 logging.debug('stderr:\n' + stderr)
@@ -919,3 +924,10 @@ class TestManager:
         )
 
         self.order += 1
+
+
+def override_tmpdir_env(old_env: Mapping, tmp_override: Path) -> Mapping:
+    new_env = dict(old_env)
+    for var in ('TMPDIR', 'TEMP', 'TMP'):
+        new_env[var] = str(tmp_override)
+    return new_env

